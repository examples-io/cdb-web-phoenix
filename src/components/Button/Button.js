import React, { Component } from 'react'
import PropTypes from 'prop-types'

/**
 * @description Button component - renders different types of buttons based on type specified via props
 */
class Button extends Component {
    render() {
        return this.renderButton();
    }

    /**
     * @description Renders the correct HTML based on the button type prop passed in.
     *              Currently supports button types: continue and primary
     *              If no button type is passed in, renders primary button
     */
    renderButton() {
        const customClasses = this.props.className ? this.props.className : '';

        if (this.props.continue) {
            return (
                <button
                    className={'button button--continue block block--centered ' + customClasses}
                    id={this.props.id}
                    onClick={this.props.onClick} >
                    <span className="button__text"> {this.props.children} </span>
                    <span className="button__icon icon icon--continue"></span>
                </button>
            );
        } else {
            // default / primary type
            return (
                <button
                    className={'button button--primary ' + customClasses}
                    id={this.props.id}
                    onClick={this.props.onClick} >
                    {this.props.children}
                </button>
            );
        }
    }
}

Button.propTypes = {
    /**
     * continue - button type
     */
    continue: PropTypes.bool,
    /**
     * primary - button type
     */
    primary : PropTypes.bool,
    /**
     * children - content to be displayed on the button, typically a string
     */
    children: PropTypes.node,
    /**
     * id - element id of the button
     */
    id: PropTypes.string,
    /**
     * onClick - function to be attached to the button's onClick event
     */
    onClick: PropTypes.func,
    /**
     * className - classes to be applied to the button element
     */
    className: PropTypes.string,
};

export default Button;