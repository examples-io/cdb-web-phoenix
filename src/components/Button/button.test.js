import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button tests', () => {
    it ('renders default button correctly', () => {
        const tree = renderer
            .create(<Button> Default </Button>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders continue button correctly', () => {
        const tree = renderer
            .create(<Button continue> Continue </Button>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('renders default button with custom class', () => {
        const tree = renderer
            .create(<Button className="test-class"> Continue </Button>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});