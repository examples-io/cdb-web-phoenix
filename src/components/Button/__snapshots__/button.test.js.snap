// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders continue button correctly 1`] = `
<button
  className="button button--continue block block--centered "
  id={undefined}
  onClick={undefined}
>
  <span
    className="button__text"
  >
     
     Continue 
     
  </span>
  <span
    className="button__icon icon icon--continue"
  />
</button>
`;

exports[`Button tests renders default button correctly 1`] = `
<button
  className="button button--primary "
  id={undefined}
  onClick={undefined}
>
   Default 
</button>
`;

exports[`Button tests renders default button with custom class 1`] = `
<button
  className="button button--primary test-class"
  id={undefined}
  onClick={undefined}
>
   Continue 
</button>
`;
